examples/features/profiling/README.md

```diff
@@ -0,0 +1,260 @@
+
+```bash
+$ go run google.golang.org/grpc/profiling/cmd \
+    -address example.com:50051                \
+    -retrieve-snapshot                        \
+    -snapshot /path/to/snapshot
+```
+
+You must provide a path to `-snapshot` that can be written to. This file will
+store the retrieved data in a raw and binary form.
+
+To process this data into a human-consumable such as
+[Catapult's trace-viewer format](https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview):
+
+```bash
+$ go run google.golang.org/grpc/profiling/cmd \
+    -address example.com:50051                \
+    -snapshot /path/to/snapshot               \
+    -stream-stats-catapult-json /path/to/json
+```
+
+This would read the data stored in `/path/to/snapshot` and process it to
+generate a JSON format that is understood by Chromium's
+[Catapult project](https://chromium.googlesource.com/catapult).
+The Catapult project comes with a utility called
+[trace-viewer](https://chromium.googlesource.com/catapult/+/HEAD/tracing/README.md),
+which can be used to generate human-readable visualizations:
+
+```bash
+$ /path/to/catapult/tracing/bin/trace2html /path/to/json --output=/path/to/html
+```
+
+When the generated `/path/to/html` file is opened with a browser, you will be
+presented with a detailed visualization of the lifecycle of all RPC requests.
+To learn more about trace-viewer and how to navigate the generated HTML, see
+[this](https://chromium.googlesource.com/catapult/+/HEAD/tracing/README.md).
+
+## Frequently Asked Questions
+
+##### I have multiple `grpc.Server`s in my application. Can I register profiling with just one of them?
+
+You may not call `profsvc.Init` more than once -- all calls except for the
+first one will return an error. As a corollary, it is also not possible to
+register or enable/disable profiling for just one `grpc.Server` or operation.
+That is, you can enable/disable profiling globally for all gRPC operations or
+none at all.
+
+##### Is `google.golang.org/grpc/profiling/cmd` the canonical implementation of a client that can talk to the profiling service?
+
+No, the command-line tool is simply provided as a reference implementation and
+as a convenience. You are free to write your own tool as long as it can
+communicate using the underlying protocol buffers.
+
+##### Is Catapult's `trace-viewer` the only option that is supported?
+
+Currently, yes. However, support for other (or better) visualization tools is
+welcome.
+
+##### What is the impact of profiling on application performance?
+
+When turned on, profiling has virtually no impact on the performance (QPS,
+latency, memory footprint) of your application. However, when turned on, expect
+a 5-10% throughput/latency penalty and double the memory footprint.
+
+Because of the negligible impact of profiling when turned off, it is
+recommended to register/initialize all applications with profiling (but leave
```

Comment:
I'm not sure if we want to "recommend".

It's not *very* useful for end users. If they are debugging performance issues, they can turn it on, but not always.

I think it's good enough to make it clear that the overhead when profiling is off is very small, so that it's OK to always register it (plus the advantages to turn on/off remotely). But don't say it's recommended.

Reply:
reworded to "you may want to"